### PR TITLE
Create .gitlab-ci.yml

### PR DESCRIPTION
**Pull request by jlaura**
## Description

The USGS software release process requires that the code be available on a .gov address. Our community is on GitHub. This PR adds a gitlab CI pipeline that can be scheduled on code.usgs.gov/astrogeology/isis that will run at some interval (every 24 hours) to keep the .gov repository up to date.
## Motivation and Context

In order to make a formal release of the ISIS code via USGS policies.

## How Has This Been Tested?

Manually, anyone with a Usgs email can access the .gov repo and see the mirrored code.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        - echo $SOURCE_REPOSITORY
+        - git config --global http.sslVerify false # Epic....
+        - git clone --mirror $SOURCE_REPOSITORY isis
+        - cd isis
+        - git remote remove origin
+        - git remote add origin $DESTINATION_REPOSITORY
+        - git push -f --prune --all
+        - git push -f --prune --tags
+    only:
+        - branches
```
Comment (acpaquette) [inline .gitlab-ci.yml:27]: Does this push all branches or just dev?

**Review by acpaquette — COMMENTED** [review]
This seems fine? I don't know if we can test it before the merge or if we need the pipeline merged to test. Small question before "Approval"

File: .gitlab-ci.yml (hunk 1 above)
Comment (jlaura) [reply .gitlab-ci.yml:27]: all branches - full on mirror (no issues or PRs)

**Review by scsides — APPROVED** [review]
Seems reasonable

**Review by acpaquette — APPROVED** [review]

**Pull request merged**